### ISSUE TITLE
belongsToMany include scope errors

### ISSUE THREAD

**Issue by ackerdev** [sequelize/sequelize#3700]
There seems to be an issue related to belongsToMany associations preventing usage of scopes with include.

Disclaimer: This repro might not be 100% correct. I don't have a way to test this at the moment, unfortunately, so this is kind of pseudo-code adapted from my codebase. Feel free to call me out if it works on your end.

### Model setup
```js
var Foo = sequelize.define({});
var Bar = sequelize.define({});
var Baz = sequelize.define({
  status: Sequelize.BOOLEAN
}, {
  scopes: {
    on: { where: { status: true } }
  }
});
 
Foo.belongsToMany(Bar);
Bar.belongsToMany(Foo);
Bar.hasMany(Baz);
Bar.hasMany(Baz.scope('on'), { as: 'on' });
Baz.belongsTo(Bar);
```

### Problem #1: Association scope does nothing
```js
Foo.find({
  include: {
    model: Bar,
    include: {
      model: Baz,
      as: 'on'
    }
  }
});
```
This seems to not work at all, ignoring the scope entirely; it never adds a where clause for `status = true`.

### Problem #2: Subquery references undefined alias
```js
Foo.find({
  include: {
    model: Bar,
    include: Baz.scope('on')
  }
});
 
Foo.find({
  include: {
    model: Bar,
    include: {
      model: Baz,
      where: {
        status: true
      }
    }
  }
});
```
Both of these cause erroneous SQL, referencing a not-yet-defined field alias. The query looks something a little like the below:

```sql
FROM (
  SELECT ...
  FROM [Foo]
  WHERE (
      SELECT ...
      FROM [Baz] AS [Foo.Bar.Baz]
      WHERE [Foo.Bar].[id] = [Foo.Bar.Baz].[bar_id]     # !!! [Foo.Bar] is not defined here
      AND [Foo.Bar.Baz].[status] = 'true'
  ) IS NOT NULL
)
```

Version: 2.1.0
Dialect: MSSQL

cc: @janmeier

**Comment by ackerdev** [comment]
Setting `required: false` on the include circumvents this issue, so it looks like scopes are implicitly making the association required.

This does not solve the issues above, but does give me a workaround for now.

**Comment by mickhansen** [comment]
Hmm yeah `include.where` sets `include.required` by default.
`association.scope` specifically doesn't do that, but there's no way to tell in your case i think.

**Comment by janmeier** [comment]
@ackerdev I fixed the first issue, and couldn't replicate the second one. Can you test on latest master :)?

**Comment by ackerdev** [comment]
Unfortunately not; the codebase has since diverged from this so I can't recreate the original problem. If that pseudo-example didn't trigger the issue, it might have been fixed in the scope refactor from 2.1.1.

All I can say is double check that you used `findOne` as opposed to `findAll`, and that you checked it on MS SQL dialect.

Otherwise, if I run across it again I'll be sure to report it, but not a lot I can do about it right now.
